File: app/src/main/java/com/example/cbanda/achieve/models/repository/GoalRepository.java
package com.example.cbanda.achieve.models.repository;

import android.arch.lifecycle.LiveData;

import com.example.cbanda.achieve.models.db.Goal;

import java.util.List;

import io.reactivex.Completable;


public interface GoalRepository {

    LiveData<List<Goal>> getListGoals();

    LiveData<Goal> getGoalById(int id);

    Completable addGoal(Goal goal);

}

File: app/src/main/java/com/example/cbanda/achieve/models/repository/GoalRepository.kt
package com.example.cbanda.achieve.models.repository

import androidx.lifecycle.LiveData

import com.example.cbanda.achieve.models.db.Goal


interface GoalRepository {

    val listGoals: LiveData<List<Goal>>

    fun getGoalById(id: Int): LiveData<Goal>

    fun addGoal(goal: Goal)

}
